src/azure-cli/azure/cli/__main__.py

```diff
@@ -91,6 +89,8 @@ def cli_main(cli, args):
                     update_all = az_cli.config.getboolean('auto-upgrade', 'all', True)
                     prompt = az_cli.config.getboolean('auto-upgrade', 'prompt', True)
                     cmd = ['az', 'upgrade', '--all', str(update_all)]
+                    az_upgrade_run = False
+                    upgrade_exit_code = None
```

Comment:
PyCharm actually already warns about the `NameError`:

![image](https://user-images.githubusercontent.com/4003950/220233395-c7896ffa-a413-46b0-b272-f3b6a0031989.png)